internal/services/sql/sql_managed_instance_administrator_resource.go

```diff
@@ -0,0 +1,210 @@
+package sql
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/v5.0/sql"
+	"github.com/gofrs/uuid"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceSqlManagedInstanceAdministrator() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSqlManagedInstanceActiveDirectoryAdministratorCreateUpdate,
+		Read:   resourceSqlManagedInstanceActiveDirectoryAdministratorRead,
+		Update: resourceSqlManagedInstanceActiveDirectoryAdministratorCreateUpdate,
+		Delete: resourceSqlManagedInstanceActiveDirectoryAdministratorDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ManagedInstanceAzureActiveDirectoryAdministratorID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"managed_instance_name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"login": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+			},
+
+			"object_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+
+			"tenant_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+
+			"azuread_authentication_only": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Computed: true,
```

Comment:
It looks like this is defaulted to false when we create it in the portal? Should we follow that same logic here?

Reply:
This is fine by me. It's better than `Computed`, question is then whether we should make the call to set it explicitly (even if `azuread_authentication_only = false`). I implemented it now in that way, that's the most secure way, but can be unnecessary.